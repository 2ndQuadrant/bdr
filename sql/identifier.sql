-- No real way to test the sysid, so ignore it
SELECT timeline= 1, dboid = (SELECT oid FROM pg_database WHERE datname = current_database())
FROM bdr.bdr_get_local_nodeid();

SELECT current_database() = 'regression';

-- Test probing for remote node information
SELECT
	r.sysid = l.sysid,
	r.timeline = l.timeline,
	r.dboid = (SELECT oid FROM pg_database WHERE datname = 'postgres'),
	variant = bdr.bdr_variant(),
	version = bdr.bdr_version(),
	version_num = bdr.bdr_version_num(),
	min_remote_version_num = bdr.bdr_min_remote_version_num(),
	is_superuser = 't'
FROM bdr.bdr_get_remote_nodeinfo('dbname=postgres') r,
     bdr.bdr_get_local_nodeid() l;

-- bdr.bdr_get_remote_nodeinfo can also be used to probe the local dsn
-- and make sure it works.
SELECT
    r.dboid = (SELECT oid FROM pg_database WHERE datname = current_database())
FROM bdr.bdr_get_remote_nodeinfo('dbname='||current_database()) r;

-- Test probing for replication connection
SELECT
	r.sysid = l.sysid,
	r.timeline = l.timeline,
	r.dboid = (SELECT oid FROM pg_database WHERE datname = 'postgres')
FROM bdr.bdr_test_replication_connection('dbname=postgres') r,
     bdr.bdr_get_local_nodeid() l;

-- Probing replication connection for the local dsn will work too
-- even though the identifier is the same.
SELECT
	r.dboid = (SELECT oid FROM pg_database WHERE datname = current_database())
FROM bdr.bdr_test_replication_connection('dbname='||current_database()) r;
