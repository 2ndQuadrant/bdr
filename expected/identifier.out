-- No real way to test the sysid, so ignore it
SELECT timeline= 1, dboid = (SELECT oid FROM pg_database WHERE datname = current_database())
FROM bdr.bdr_get_local_nodeid();
 ?column? | ?column? 
----------+----------
 t        | t
(1 row)

SELECT current_database() = 'regression';
 ?column? 
----------
 t
(1 row)

-- Test probing for remote node information
SELECT
	r.sysid = l.sysid,
	r.timeline = l.timeline,
	r.dboid = (SELECT oid FROM pg_database WHERE datname = 'postgres'),
	variant = bdr.bdr_variant(),
	version = bdr.bdr_version(),
	version_num = bdr.bdr_version_num(),
	min_remote_version_num = bdr.bdr_min_remote_version_num(),
	is_superuser = 't'
FROM bdr.bdr_get_remote_nodeinfo('dbname=postgres') r,
     bdr.bdr_get_local_nodeid() l;
 ?column? | ?column? | ?column? | ?column? | ?column? | ?column? | ?column? | ?column? 
----------+----------+----------+----------+----------+----------+----------+----------
 t        | t        | t        | t        | t        | t        | t        | t
(1 row)

-- bdr.bdr_get_remote_nodeinfo can also be used to probe the local dsn
-- and make sure it works.
SELECT
    r.dboid = (SELECT oid FROM pg_database WHERE datname = current_database())
FROM bdr.bdr_get_remote_nodeinfo('dbname='||current_database()) r;
 ?column? 
----------
 t
(1 row)

-- Test probing for replication connection
SELECT
	r.sysid = l.sysid,
	r.timeline = l.timeline,
	r.dboid = (SELECT oid FROM pg_database WHERE datname = 'postgres')
FROM bdr.bdr_test_replication_connection('dbname=postgres') r,
     bdr.bdr_get_local_nodeid() l;
 ?column? | ?column? | ?column? 
----------+----------+----------
 t        | t        | t
(1 row)

-- Probing replication connection for the local dsn will work too
-- even though the identifier is the same.
SELECT
	r.dboid = (SELECT oid FROM pg_database WHERE datname = current_database())
FROM bdr.bdr_test_replication_connection('dbname='||current_database()) r;
 ?column? 
----------
 t
(1 row)

